src/me/drton/jmavsim/SimpleSensors.java

```diff
@@ -157,7 +163,32 @@ public LatLonAlt getGlobalPosition() {
     public void setGlobalPosition(Vector3d pos) {
         if (pos == null)
             pos = object.getPosition();
-        globalPosition = globalProjector.reproject(new double[]{pos.x, pos.y, pos.z});
+
+        long t = System.currentTimeMillis();
+        double dt = 0.0f;
+        if ( prevUpdateTime > 0 ) {
+            dt = (t - this.prevUpdateTime) * 1e-3f;
+        }
+
+        // add noise (random walk)
+        if ( dt > 0.0f ) {
+            double sqrtDt = java.lang.Math.sqrt(dt);
+            double noiseX = sqrtDt*randomNoise(gpsNoiseStdDev);
+            double noiseY = sqrtDt*randomNoise(gpsNoiseStdDev);
+            double noiseZ = sqrtDt*randomNoise(gpsNoiseStdDev);
+
+            this.randomWalkGpsX += noiseX;
```

Comment:
Can you add the correlation here, like https://github.com/PX4/sitl_gazebo/pull/82/files#diff-af95b4649436828244349b1766ab5c46R530 and the integration dt?
Like:
```
this.randomWalkGpsX += noiseX * dt - this.randomWalkGpsX / gpsCorrelationTime;
```

Reply:
Thank you for your advice . I will add correlation.
BTW, could you tell me some reference paper for the correlation?.  
First time, I copy your code with correlation. However I don't understand that. So I remove it.
Actually, I don't understand why the correlation is added.

Reply:
The correlation is that the position does not arbitrarily drift away (which in reality does not happen either). @jgoppert has looked into this more extensively and knows more than I do.